explore.test.ts

```diff
@@ -0,0 +1,13 @@
+import { Directions } from "./contants"
+import explore from "./explore"
+import Plateau from "./plateau"
+
+it('Should return latest position of rover', () => {
+    const plateau = new Plateau(4,4)
+    expect(explore(2,3,Directions.N,'LMRMRMRMRMRM', plateau)).toEqual({x: 1, y: 4, direction: Directions.N})
+})
+
+it('Should throw error if position out of range', () => {
```

Comment:
Are you sure you have followed TDD?
There are too many ifs and elses in the source code and I assume one test doesn't cover all of them.

Reply:
The if else logic in explore simply updates the x, y, and direction. The instruction string in test should cover all directions and so the coverage here is 100%. I didn't see any extra logic in explore to split the test.

Reply:
But the string is "LMRMRMRMRMRM", doesn't cover all the blocks as I understand?

Reply:
I see that it doesn't cover 1 else inside of L's condition. Other than that it covers all.

Not sure why branch coverage shows 100.

![Screenshot from 2023-09-29 15-47-16](https://github.com/piyush-debuide/mars_rover/assets/106428525/ffbf6a31-9b78-484a-afcc-2e37cf94e6d5)